dvc/exceptions.py

```diff
@@ -344,3 +344,13 @@ def __init__(self, path, external_repo_path, external_repo_url):
 class HTTPError(DvcException):
     def __init__(self, code, reason):
         super(HTTPError, self).__init__("'{} {}'".format(code, reason))
+
+
+class TooManyOpenFilesError(DvcException):
+    def __init__(self, cause):
+        super(TooManyOpenFilesError, self).__init__(
+            "Operation failed due to too many open file descriptors. reduce "
+            "the number of jobs or increase open file descriptors limit to "
+            "prevent this.",
```

Comment:
We should put the emphasis on "increase your ulimit", as that is the most reasonable approach. Also, it is better if we tell how to do it or refer to a doc.